PennMobile/src/main/java/com/pennapps/labs/pennmobile/BookGsrFragment.kt

```diff
@@ -0,0 +1,142 @@
+package com.pennapps.labs.pennmobile
+
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.EditText
+import android.widget.Toast
+import butterknife.ButterKnife
+import com.pennapps.labs.pennmobile.api.Labs
+import kotlinx.android.synthetic.main.gsr_details_book.view.*
+import retrofit.Callback
+import retrofit.RetrofitError
+import retrofit.client.Response
+import java.io.BufferedReader
+import java.io.IOException
+import java.io.InputStreamReader
+
+
+class BookGsrFragment : Fragment() {
+
+    // fields for booking
+    internal var firstName: EditText? = null
+    internal var lastName: EditText? = null
+    internal var email: EditText? = null
```

Comment:
Is it possible for these firstName, lastName, email variables to be null? Having all these nullable values makes some of the code pretty clunky.